code/modules/antagonists/blood_worm/actions/cocoon.dm

```diff
@@ -0,0 +1,333 @@
+	name = "Mature"
+	desc = "Enter incubation in a cocoon, emerging as a juvenile blood worm."
+
+	button_icon_state = "mature_hatchling"
+
+	cocoon_type = /obj/structure/blood_worm_cocoon/hatchling
+	new_worm_type = /mob/living/basic/blood_worm/juvenile
+
+	total_blood_required = 1000
+
+/datum/action/cooldown/mob_cooldown/blood_worm/cocoon/hatchling/Activate(atom/target)
+	if (tgui_alert(owner, "Are you sure? After [cocoon_time / 10] seconds, you will become a juvenile, gaining stat increases and the ability to spit corrosive blood, but losing the ability to ventcrawl.", "Mature", list("Yes", "No"), 30 SECONDS) != "Yes")
+		return
+
+	return ..()
+
+/obj/structure/blood_worm_cocoon/hatchling
+	name = "small blood cocoon"
+	desc = "The incubation cocoon of a hatchling blood worm. Its surface is slowly shifting."
+
+	icon_state = "cocoon-small"
+
+	max_integrity = 100
+	damage_deflection = 10
+
+/obj/structure/blood_worm_cocoon/hatchling/examine(mob/user)
+	return ..() + span_warning("It can be broken to prevent the blood worm from maturing.")
+
+/datum/action/cooldown/mob_cooldown/blood_worm/cocoon/juvenile
+	name = "Mature"
+	desc = "Enter incubation in a cocoon, emerging as an adult blood worm."
+
+	button_icon_state = "mature_juvenile"
+
+	cocoon_type = /obj/structure/blood_worm_cocoon/juvenile
+	new_worm_type = /mob/living/basic/blood_worm/adult
+
+	total_blood_required = 2500
+
+/datum/action/cooldown/mob_cooldown/blood_worm/cocoon/juvenile/Activate(atom/target)
+	if (tgui_alert(owner, "Are you sure? After [cocoon_time / 10] seconds, you will become an adult, gaining stat increases and the ability to spit bursts of corrosive blood by right-clicking with Spit Blood while outside of a host.", "Mature", list("Yes", "No"), 30 SECONDS) != "Yes")
+		return
+
+	return ..()
+
+/obj/structure/blood_worm_cocoon/juvenile
+	name = "medium blood cocoon"
+	desc = "The incubation cocoon of a juvenile blood worm. Its surface is slowly shifting."
+
+	icon_state = "cocoon-medium"
+
+	max_integrity = 150
+	damage_deflection = 15
+
+/obj/structure/blood_worm_cocoon/juvenile/examine(mob/user)
+	return ..() + span_warning("It can be broken to prevent the blood worm from maturing, but it looks rather tough.")
+
+/datum/action/cooldown/mob_cooldown/blood_worm/cocoon/adult
+	name = "Reproduce"
+	desc = "Enter incubation in a cocoon, creating 5 new hatchlings including yourself."
+
+	button_icon_state = "reproduce"
+
+	cocoon_type = /obj/structure/blood_worm_cocoon/adult
+	new_worm_type = /mob/living/basic/blood_worm/hatchling
+
+	total_blood_required = 0
+
+	var/num_hatchlings = 4 // in addition to the original
+
+	var/list/candidates = null
+
+/datum/action/cooldown/mob_cooldown/blood_worm/cocoon/adult/Activate(atom/target)
+	if (tgui_alert(owner, "Are you sure? After [cocoon_time / 10] seconds, you will create [num_hatchlings + 1] new hatchlings, including yourself.", "Reproduce", list("Yes", "No"), 30 SECONDS) != "Yes")
+		return
```

Comment:
I feel like you need to check for input stalling for this otherwise someone could spam the heck out of it

Reply:
I added a check after each one. The check before comes implicitly, as Activate() is only reached if IsAvailable() returns TRUE. This should prevent spamming ghosts, duplicating cocoons, etc.